lib/Test/Stream/Formatter/TAP.pm

```diff
@@ -99,6 +116,193 @@ sub _autoflush {
     select $old_fh;
 }
 
+sub event_tap {
+    my $self = shift;
+    my ($e, $num) = @_;
+
+    my $converter = $CONVERTERS{ref($e)};
+
+    $num = undef if $self->{+NO_NUMBERS};
+
+    # Legacy Support for $e->to_tap
+    unless ($converter) {
+        my $legacy = $e->can('to_tap') or return;
+        return if $legacy == \&Test::Stream::Event::to_tap;
```

Comment:
Need a comment here really explaining that calling the legacy `to_tap` method will cause infinite recursion, which is why you refuse to call it if it hasn't been overloaded.

Probably also worth mentioning that if anyone implemented `to_tap` as

```perl

sub to_tap {
    ...
    $self->SUPER::to_tap(@args);
}
```
You can now expect the infinite recursion to come back.